CRINF-206: Work-in-progress bundling functionalitly

pbitty opened the pull request:
This is a staging area for me building out the bundling functionality into go-task.  Once it's stable and ready, I will open another PR against [go-task/task](https://github.com/go-task/task).

[CRINF-206]

[CRINF-206]: https://wattpad.atlassian.net/browse/CRINF-206?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

pbitty committed:
Add remote taskfile HTTP tests

pbitty committed:
Add go-getter implementation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Wattpad/go-task?pullRequest=1) <br/>All committers have signed the CLA.

pbitty committed:
Add caching support for remote directories

pbitty committed:
Add test data

This data will be used by tests in the following commit, which will
clone this branch to test git remotes.

pbitty committed:
Add git remote tests

pbitty committed:
Disable remote caching tests

pbitty committed:
TEMP commit: applying caching refactor from #1771

pbitty committed:
Make source location consistent between cached and uncached nodes

BanglanfengPan committed:
add version number

BanglanfengPan committed:
update the release

BanglanfengPan committed:
Revert "update the release"

This reverts commit 9208586b7da20e1ec98ccdc1bd661c5db707badf.

BanglanfengPan committed:
Revert "add version number"

This reverts commit 0854782ec5e5c8b8090f1a82010a4287ab731f04.
